Sources/web3swift/Web3/Web3+Structures.swift

```diff
@@ -110,12 +110,20 @@ extension EthereumTransaction:Decodable {
         case r
         case s
         case value
+        case type  // present in EIP-1559 transaction objects
     }
     
     public init(from decoder: Decoder) throws {
         let options = try TransactionOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
         
+        // test to see if it is a EIP-1559 wrapper
+        let envelope = try decodeHexToBigUInt(container, key: .type, allowOptional: true)
+        if( (envelope != nil) && (envelope! != BigInt(0)) ) {
```

Comment:
we should not be using force unwrap

Reply:
sure I can change this to an if let or guard